AudiusProject/audius-protocol
# [AUD-190] - Making Mad-Dog more robust

### Description
_What is the purpose of this PR? What is the current behavior? New behavior? Relevant links (e.g. Trello) and/or information pertaining to PR?_

**Purpose:** 
Mad-dog is our fantastic integration tool that allows us to programmatically test our system. However, it is lacking some optimizations, like speed and useful logs. This PR's intention is to:
- Eliminate manual timeouts, instead opting for polling based on tx blocknumbers
- Cache CI dependencies for faster builds
- Consolidate test scenarios 
- Add better logging to display stack trace and useful information (I.e. wallet addresses, user ids, clock values)
- Minor refactors (styling, comments, etc)
- Update branch tests by auto-running mad-dog build with every new PR and new commit to a PR that does not have the name `master` or `hotfix` in branch title [(see test)](https://github.com/AudiusProject/audius-protocol/pull/1200)

Also included in this PR is:
- A route called `/ipld_block_check` to check the max block indexed in the ipld block table (used to poll the ipld blocks in mad-dog)
- A configurable env var called `audius_discprov_blacklist_block_indexing_window` to set the ipld block indexing interval (unit is in seconds)

TODO (not for the scope of this PR / future task): 
- Add a wallet check via disc prov to see if wallet has been registered
### Tests
